src/index.js

```diff
@@ -1,14 +1,135 @@
-import _ from 'lodash';
 import './style.css';
 
-function component() {
-  const element = document.createElement('div');
+const form = document.querySelector('.todo-add');
+const input = document.querySelector('.form-field');
+const ul = document.getElementById('todo-list');
+let li;
 
-  // Lodash, now imported by this script
-  element.innerHTML = _.join(['Hello', 'ahmed'], ' ');
-  element.classList.add('hello');
+let tasks = [];
 
-  return element;
+function getTasks() {
+  if (localStorage.getItem('tasks') == null) {
+    tasks = [];
+  } else {
+    tasks = Array.from(JSON.parse(localStorage.getItem('tasks')));
+  }
+  return tasks;
 }
 
-document.body.appendChild(component());
+class Task {
+  constructor(description, index) {
+    this.description = description;
+    this.completed = false;
+    this.index = index;
+  }
+
+  set updatedIndex(newIndex) {
+    this.index = newIndex;
+  }
+
+  set updatedDesc(newDesc) {
+    this.description = newDesc;
+  }
+}
+
+function prepareEdit(task, btn) {
+  btn.addEventListener('click', () => {
+    const el = document.querySelector(`[data-index="${task.index}"]`);
+    const p = el.children[1];
+    const input = document.createElement('input');
+    input.type = 'text';
+    input.value = p.textContent;
+    input.id = task.index;
+    el.insertBefore(input, p);
+    el.removeChild(p);
+    input.focus();
+    input.select();
+    input.classList.add('edit');
+  });
+}
```

Comment:
Your implementation for editing a task is working like a charm 💪😎

### OPTIONAL
To edit a task you need to click on the three dots(Works perfectly 👀) but it would enhance a lot the user experience if you would toggle editing if the user clicks or double clicks on the task 🤓